### PR TITLE
[Task 1841825] add resource group in resource lib

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        final ResourceId id = ResourceId.fromString(idStr);
+        return get(id.subscriptionId(), id.resourceGroupName());
+    }
+
+    public ResourceGroupEntity getByName(@Nonnull String name) {
+        return get(getDefaultSubscription().getId(), name);
+    }
+
+    public ResourceGroupEntity get(@Nonnull String subscriptionId, @Nonnull String name) {
+        return ResourceGroupEntity.fromResource(getAzureResourceManager(subscriptionId).resourceGroups().getByName(name));
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java:57]: As sdk will throw exception for none exist resources, do we need to handle this exception and return null here? Or do we need to add an `exist()` method to determine whether resource group with specific name exists?

File: azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    public ResourceGroupEntity getById(@Nonnull String idStr) {
+        final ResourceId id = ResourceId.fromString(idStr);
+        return get(id.subscriptionId(), id.resourceGroupName());
+    }
+
+    public ResourceGroupEntity getByName(@Nonnull String name) {
+        return get(getDefaultSubscription().getId(), name);
+    }
+
```
[the remark this answers is not in the document]
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java:55]: will add in future

File: azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java (hunk 1 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java:57]: throw exception will be fine, null for non-existing is to abort exception handling, we don't have the need to check exist?

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java:57]: will add exist later

File: azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import org.apache.commons.lang3.StringUtils;
+import reactor.core.publisher.Flux;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public class AzureResourceGroup extends SubscriptionScoped<AzureResourceGroup> implements AzureService {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-resource-lib/src/main/java/com/microsoft/azure/toolkit/lib/resource/AzureResourceGroup.java:29]: rename to AzureGroup

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**